crates/spark/src/services/transfer.rs

```diff
@@ -675,35 +701,82 @@ impl TransferService {
-                let leaves_to_receive = leaves_to_receive.clone();
-
-                let task = async move {
-                    operator
-                        .client
-                        .claim_transfer_tweak_keys(
-                            operator_rpc::spark::ClaimTransferTweakKeysRequest {
-                                transfer_id: transfer.id.to_string(),
-                                owner_identity_public_key: identity_public_key,
-                                leaves_to_receive,
-                            },
-                        )
-                        .await
-                        .map_err(|e| {
-                            if let OperatorRpcError::Connection(status) = &e
-                                && status.code() == Code::AlreadyExists
-                            {
-                                return ServiceError::TransferAlreadyClaimed;
-                            }
-
-                            e.into()
-                        })
+            let Some(leaves_to_receive) = leaves_tweaks_map.get(&operator.identifier) else {
+                continue;
+            };
+
+            let identity_public_key = self.signer.get_identity_public_key()?.serialize().to_vec();
+            let leaves_to_receive = leaves_to_receive.clone();
+
+            let task = async move {
+                let result = operator
+                    .client
+                    .claim_transfer_tweak_keys(operator_rpc::spark::ClaimTransferTweakKeysRequest {
+                        transfer_id: transfer.id.to_string(),
+                        owner_identity_public_key: identity_public_key,
+                        leaves_to_receive,
+                    })
+                    .await;
+
+                let err = match result {
+                    Ok(()) => {
+                        trace!(
+                            "Operator {} successfully applied key tweaks for transfer {}",
+                            operator.identity_public_key, transfer.id
+                        );
+                        return Ok::<_, ServiceError>(());
+                    }
+                    Err(e) => e,
                 };
-                tasks.push(task);
-            }
+
+                let OperatorRpcError::Connection(status) = err else {
+                    return Err(err.into());
+                };
+
+                debug!(
+                    "Operator {} returned connection error: {:?}",
+                    operator.identity_public_key, status
+                );
+
+                // There was an error tweaking the keys. It's likely the transfer was not in state SenderKeyTweaked after all.
+                // Let's find out what state we're in.
+                // Fetch the leaf remotely from the operator. Determine its state, and return info about it to the caller.
+                let operator_transfers = operator
+                    .client
+                    .query_all_transfers(TransferFilter {
```

Comment:
Why don't we use the `self.query_transfer` function like we do in `claim_transfer_sign_refunds`?

Reply:
Because that queries the coordinator, while here we specifically want to know the state of the transfer on that coordinator.

Reply:
@JssDWt  I see makes sense. A general question about the approach: We refresh the transfer inside the more high level `claim_transfer` function. We have a retry mechanism there and we could check if the transfer has already claimed.  Just wondering if you considered that and if you see advantages on the current approach.

Reply:
Thanks Jesse!
I think you are right. If we start the process with tweaks only in some of the operators  (whether we exited in the middle of the process or another client  was able to twek some);we must query the leaves in this operator and skip if already tweaked.
Thanks for clarifying.